ha/resource/vip_health_monitor.py

```diff
@@ -0,0 +1,146 @@
+
+from ha.core.config.config_manager import ConfigManager
+from ha.resource.resource_agent import CortxServiceRA
+from ha.execute import SimpleCommand
+from ha import const
+
+class VipHealthMonitor(CortxServiceRA):
+    """
+    Check VIP health check up and failover if unhealthy.
+    """
+    def __init__(self):
+        """
+        Initialize the class.
+        """
+        super(VipHealthMonitor, self).__init__()
+        self._execute = SimpleCommand()
+
+    @staticmethod
+    def metadata() -> int:
+        """
+        Provide meta data for resource agent and parameter
+        """
+        env: str =r"""<?xml version="1.0"?>
+        <!DOCTYPE resource-agent SYSTEM "ra-api-1.dtd">
+        <resource-agent name="vip_health_check">
+        <version>1.0</version>
+
+        <longdesc lang="en">
+            Check health of vip.
+        </longdesc>
+        <shortdesc lang="en">Health Check</shortdesc>
+        <parameters>
+        <parameter name="vip" required="1">
+        <longdesc lang="en"> VIP that need monitor </longdesc>
+        <shortdesc lang="en"> VIP </shortdesc>
+        <content type="string"/>
+        </parameter>
+        <parameter name="nic" required="1">
+        <longdesc lang="en"> NIC interface </longdesc>
+        <shortdesc lang="en"> NIC interface </shortdesc>
+        <content type="string"/>
+        </parameter>
+        </parameters>
+        <actions>
+        <action name="start"        timeout="3s" />
+        <action name="stop"         timeout="3s" />
+        <action name="monitor"      timeout="3s" interval="60s" depth="0" />
+        <action name="meta-data"    timeout="4s" />
+        </actions>
+        </resource-agent>
+        """
+        sys.stdout.write(env)
+        return const.OCF_SUCCESS
+
+    def start(self) -> int:
+        """
+        Start monitoring.
+        """
+        rc: int = self.monitor()
+        Log.info(f"start action on vip monitoring with rc: {rc}")
+        return rc
+
+    def stop(self) -> int:
+        """
+        Stop Vip.
+        """
+        return const.OCF_SUCCESS
+
+    def monitor(self) -> int:
+        """
+        Monitor vip
+        """
+        res_param = self.get_env()
+        vip: str = res_param["OCF_RESKEY_vip"]
+        nic: str = res_param["OCF_RESKEY_nic"]
+        output, error, rc = self._execute.run_cmd(f"ip a s {nic}")
+        if rc != 0:
+            Log.error(f"Failed to get ip address for {nic} with error {error}")
+            return const.OCF_ERR_GENERIC
+        status_str = output.split("\n")[0].split(" ")
+        status = status_str[status_str.index("state") + 1]
+        if status != "UP":
+            Log.error(f"VIP Health failed, {nic} is down")
+            return const.OCF_ERR_GENERIC
+        ip_list = []
+        for line in output.split("\n"):
+            if len(line.split()) > 0 and "inet" in line.split()[0] and vip not in line.split():
+                ip = line.split()[1].split("/")[0]
```

Comment:
do we need these two lines ? we just need to know that the len is non zero right ?

Reply:
Yes there are two case
1. ifdown etho
(generic with all interface) after this i have seen that state is not changing to DOWN but ips are getting disappered and no ip will be assigned to interface
which need above two line

2.ip link set down eth0
Here state is changing to DOWN and so, not reaching in for loop
```
        if status != "UP":
            Log.error(f"VIP Health failed, {nic} is down")
            return const.OCF_ERR_GENERIC
```
is used in second case